src/accounts/index.ts

```diff
@@ -239,166 +244,32 @@ async function deploy(
-  }
-  // No need to deploy manually for the intent flow
-}
-
-async function deployStandalone(chain: Chain, config: RhinestoneAccountConfig) {
-  const deployer = config.deployerAccount
-  if (deployer) {
-    return deployStandaloneWithEoa(chain, config, deployer)
-  }
-  return deployStandaloneWithBundler(chain, config)
-}
-
-async function deploy7702Self(chain: Chain, config: RhinestoneAccountConfig) {
-  if (!config.eoa) {
-    throw new Eip7702AccountMustHaveEoaError()
-  }
-
-  const account = getAccountProvider(config)
-  const { implementation, initializationCallData } = getDeployArgs(config)
-  if (!initializationCallData) {
-    throw new Error(
-      `Initialization call data not available for ${account.type}`,
-    )
-  }
-
-  const publicClient = createPublicClient({
-    chain,
-    transport: createTransport(chain, config.provider),
-  })
-  const accountClient = createWalletClient({
-    account: config.eoa,
-    chain,
-    transport: createTransport(chain, config.provider),
-  })
-
-  const authorization = await accountClient.signAuthorization({
-    contractAddress: implementation,
-    executor: 'self',
-  })
-
-  const hash = await accountClient.sendTransaction({
-    chain,
-    authorizationList: [authorization],
-    to: config.eoa.address,
-    data: initializationCallData,
-  })
-  await publicClient.waitForTransactionReceipt({ hash })
-}
-
-async function deployStandaloneWithEoa(
-  chain: Chain,
-  config: RhinestoneAccountConfig,
-  deployer: Account,
-) {
-  const { factory, factoryData } = getDeployArgs(config)
-  const publicClient = createPublicClient({
-    chain: chain,
-    transport: createTransport(chain, config.provider),
-  })
-  const client = createWalletClient({
-    account: deployer,
-    chain: chain,
-    transport: createTransport(chain, config.provider),
-  })
-  const tx = await client.sendTransaction({
-    to: factory,
-    data: factoryData,
-  })
-  await publicClient.waitForTransactionReceipt({ hash: tx })
-}
-
-async function deployStandaloneWithBundler(
-  chain: Chain,
-  config: RhinestoneAccountConfig,
-) {
+async function deployWithIntent(chain: Chain, config: RhinestoneAccountConfig) {
   const publicClient = createPublicClient({
     chain,
     transport: createTransport(chain, config.provider),
   })
-  const bundlerClient = getBundlerClient(config, publicClient)
-  const smartAccount = await getSmartAccount(config, publicClient, chain)
-  const { factory, factoryData } = getDeployArgs(config)
-  const opHash = await bundlerClient.sendUserOperation({
-    account: smartAccount,
-    factory,
-    factoryData,
+  const address = getAddress(config)
+  const code = await publicClient.getCode({ address })
+  if (code) {
+    // Already deployed
+    return
+  }
+  await sendTransaction(config, {
+    targetChain: chain,
     calls: [
       {
-        to: zeroHash,
-        value: 0n,
+        to: zeroAddress,
         data: '0x',
       },
     ],
   })
```

Comment:
Btw this is not gonna work on Optimism for now, as the default `tokenRequest` is 1 wei of ETH